src/Kiota.Builder/Writers/Go/CodeMethodWriter.cs

```diff
@@ -0,0 +1,468 @@
+                    break;
+                case CodeMethodKind.ClientConstructor:
+                    WriteConstructorBody(parentClass, codeElement, writer, inherits);
+                    WriteApiConstructorBody(parentClass, codeElement, writer);
+                    writer.WriteLine("return m");
+                break;
+                case CodeMethodKind.Constructor:
+                    WriteConstructorBody(parentClass, codeElement, writer, inherits);
+                    writer.WriteLine("return m");
+                    break;
+                case CodeMethodKind.RequestBuilderBackwardCompatibility:
+                    WriteRequestBuilderBody(parentClass, codeElement, writer);
+                    break;
+                default:
+                    writer.WriteLine("return nil");
+                break;
+            }
+            writer.DecreaseIndent();
+            writer.WriteLine("}");
+        }
+
+        private void WriteRequestBuilderBody(CodeClass parentClass, CodeMethod codeElement, LanguageWriter writer)
+        {
+            var importSymbol = conventions.GetTypeString(codeElement.ReturnType, parentClass);
+            var currentPathProperty = codeElement.Parent.GetChildElements(true).OfType<CodeProperty>().FirstOrDefault(x => x.IsOfKind(CodePropertyKind.CurrentPath));
+            conventions.AddRequestBuilderBody(currentPathProperty != null, importSymbol, writer);
+        }
+
+        private void WriteSerializerBody(CodeClass parentClass, LanguageWriter writer) {
+            var additionalDataProperty = parentClass.GetPropertiesOfKind(CodePropertyKind.AdditionalData).FirstOrDefault();
+            var shouldDeclareErrorVar = true;
+            if(parentClass.StartBlock is CodeClass.Declaration declaration &&
+                declaration.Inherits != null) {
+                writer.WriteLine($"err := m.{declaration.Inherits.Name.ToFirstCharacterUpperCase()}.Serialize(writer)");
+                WriteReturnError(writer);
+                shouldDeclareErrorVar = false;
+            }
+            foreach(var otherProp in parentClass.GetPropertiesOfKind(CodePropertyKind.Custom)) {
+                WriteSerializationMethodCall(otherProp.Type, parentClass, otherProp.SerializationName ?? otherProp.Name.ToFirstCharacterLowerCase(), $"m.Get{otherProp.Name.ToFirstCharacterUpperCase()}()", shouldDeclareErrorVar, writer);
+                if(shouldDeclareErrorVar)
+                    shouldDeclareErrorVar = false;
+            }
+            if(additionalDataProperty != null) {
+                writer.WriteLine($"err {errorVarDeclaration(shouldDeclareErrorVar)}= writer.WriteAdditionalData(m.Get{additionalDataProperty.Name.ToFirstCharacterUpperCase()}())");
+                WriteReturnError(writer);
+            }
+            writer.WriteLine("return nil");
+        }
+        private static string errorVarDeclaration(bool shouldDeclareErrorVar) => shouldDeclareErrorVar ? ":" : string.Empty;
```

Comment:
Out of curiousity, do you mark some of the member methods as static mainly performance reasons? Or does it help you identify a case when a function is not touching instance data

Reply:
it's a suggestion from sonarqube. https://rules.sonarsource.com/csharp/RSPEC-2325